[18.0][MIG] web_assets_warmup: Migration to 18.0

xaviedoanhduy opened the pull request:
- `commit_assetsbundle` was deleted when rendering the report in commit: https://github.com/odoo/odoo/commit/ad4deba
- `declare assets in manifest, not in xml` so this commit changed xmlid parameter to bunble, see in: https://github.com/odoo/odoo/pull/60632/commits/8cc06617

SilvioC2C committed:
[14.0] web_m2x_options_manager: New module to manage create/edit options from interface

Tardo committed:
[ADD] web_time_range_menu_custom

Tardo committed:
[FIX] web_time_range_menu_custom: Mix custom and 'no custom' periods

Steps to reproduce:

- Go to pivot view
- Select a custom time range
- Select a predefined comparison period (Ex. Previous Period)

Now you see that the comparison period is not being applied.
After this commit the comparison period is applied correctly.

TT29249

oca-travis committed:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[UPD] README.rst

yankinmax committed:
[UPD] web_m2x_options_manager: Add option to prevent Create/Edit wizard to pop up

oca-travis committed:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
web_m2x_options_manager 14.0.1.1.0

fkantelberg committed:
[ADD][16.0] Module web_dark_mode

Commit:
[UPD] Update web_dark_mode.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

GoodERPJeff committed:
Added translation using Weblate (Chinese (zh))

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/es_AR/

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: black, isort, prettier

fkantelberg committed:
[FIX] web_dark_mode: Fix uninstallation

OCA-git-bot committed:
web_dark_mode 16.0.1.0.1

badbole committed:
Added translation using Weblate (Croatian)

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/hr/

tate11 committed:
Added translation using Weblate (Turkish)

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/tr/

francesco-ooops committed:
Added translation using Weblate (Italian)

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 76.4% (26 of 34 strings)

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/it/

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 15.0

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: Dates include today but other granularities not

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

jbaudoux committed:
[ADD] web_assets_warmup

mt-software-de committed:
[IMP] web_assets_warmup: Add database lock in post_load_hook

With dedicated services for odoo web and odoo cron
we can get a serialize access error on startup

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/

gregory-moka committed:
Added translation using Weblate (French)

gregory-moka committed:
Translated using Weblate (French)

Currently translated at 50.0% (2 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/fr/

HekkiMelody committed:
[FIX] all: fix pre-commit new checks

Commit:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/

fkantelberg committed:
[FIX] web_dark_mode: Add fields to field lists

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: web-16.0/web-16.0-web_dark_mode
Translate-URL: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_dark_mode/it/

yajo committed:
[ADD] web_touchscreen: enable kanban sub-form views for tablets

This module prefers kanban sub-form views when using a tablet, or any other
device with a touch screen, no matter its size.

@moduon MT-4472

Commit:
[UPD] Update web_touchscreen.pot

yajo committed:
[DOCS] web_touchscreen: add screenshot

Follow suggestion from https://github.com/OCA/web/pull/2701#issuecomment-1870681682.

@moduon MT-4472

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update web_assets_warmup.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 16.0

mymage committed:
Added translation using Weblate (Italian)

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: pre-commit auto fixes

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 17.0

mymage committed:
Added translation using Weblate (Italian)

CarlosRoca13 committed:
[IMP] web_time_range_menu_custom: pre-commit auto fixes

CarlosRoca13 committed:
[MIG] web_time_range_menu_custom: Migration to 18.0

natuan9 committed:
[IMP] web_datetime_picker_default_time: pre-commit auto fixes

natuan9 committed:
[MIG] web_datetime_picker_default_time: Migration to 18.0

xaviedoanhduy committed:
[IMP] web_assets_warmup: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] web_assets_warmup: Migration to 18.0

nilshamerlinck commented:
isn't `web_assets_warmup` useless since `https://github.com/odoo/odoo/pull/131353`?

xaviedoanhduy commented:
> https://github.com/odoo/odoo/pull/131353

as per the purpose of this module from https://github.com/OCA/web/pull/2614#issue-1898108560, this module will help assets to be are well-generated when odoo starts, not when the report is printed.
from commit https://github.com/odoo/odoo/pull/131353/commits/5d8c2fd3 the crash when printing the report seems to be improved. but I still migration this module to preload the assets instead of rendering it when I press print on the report

nilshamerlinck commented on web_assets_warmup/models/ir_actions_report.py:
```diff
@@ -0,0 +1,30 @@
+    def cron_generate_assets(self):
+        """Ensure that the assets are well-generated in the database."""
+        logger.info("Ensure that assets are generated and stored in the database...")
+        # Call `_get_asset_nodes` as done when printing a report based on
+        # `web.report_layout` template (used by `web.html_container`)
```
```suggestion
```

nilshamerlinck replied:
outdated (to replace by appropriate comment)

nilshamerlinck commented on web_assets_warmup/models/ir_actions_report.py:
```diff
@@ -0,0 +1,28 @@
+        logger.info("Ensure that assets are generated and stored in the database...")
+        bundles = [
+            "web.report_assets_common",
+            "web.assets_common",
```
removed: see `https://github.com/odoo/odoo/pull/132190`

nilshamerlinck commented:
Could you improve PR description based on these additions?

- Since odoo v17.0, assets are not generated anymore in the same transaction that generates a report. They are generated when retrieved, through the /web/assets route. See `https://github.com/odoo/odoo/pull/140124`
- As a consequence, the problematic case that lead to the creation of `web_assets_warmup` can't occure anymore: `https://github.com/OCA/web/pull/2614#issue-1898108560`
- That said, module still has some value in "warming" up common assets for reports
- Note that since 16.0, odoo introduced a more generic `env['ir.qweb']._pregenerate_assets_bundles()`, but used for tests only, see `https://github.com/odoo/odoo/pull/99176

nilshamerlinck commented:
> That said, module still has some value in "warming" up common assets for reports

did you do some tests to see how much it saves on first report generation?

xaviedoanhduy commented:
> did you do some tests to see how much it saves on first report generation?


hi @nilshamerlinck, thanks for the suggestions on how to improve the PR description.

I did some speed tests when printing the report for the first time (check executed when calling `_render_qweb_pdf` method)([ref](https://github.com/odoo/odoo/blob/730247f96611813cf65cd30a237d29d58b568a96/odoo/addons/base/models/ir_actions_report.py#L1001C9-L1001C25))

- when installing the `sale_management` module, and not installing `web_assets_warmup`: 3.861 (s)
![image](https://github.com/user-attachments/assets/73962520-9808-4495-ad46-13a815638302)

- when installing the `sale_management` and `web_assets_warmup` modules but not manually executing the schedule action task (Generate report assets): 3.949 (s)
![image](https://github.com/user-attachments/assets/d573ab3d-5282-4ecc-bbdc-08b4bf1f61d0)

- when installing the `sale_management` and `web_assets_warmup` modules then manually executing the schedule action task (Generate report assets): 2.860 (s)
![image](https://github.com/user-attachments/assets/ac4ab220-38dc-4660-9880-1b3297ab22a8)

nilshamerlinck commented:
ok thanks, these are interesting metrics that you can add in the PR description as well
I'm not sure the module is useful for 1s 1 time gain, but we'll let them judge

nilshamerlinck commented:
(I see some typos in the PR description on OCA side, please have a look)

xaviedoanhduy commented:
> (I see some typos in the PR description on OCA side, please have a look)

I have updated it, thank you

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (33 of 33 strings)

Translation: web-14.0/web-14.0-web_m2x_options_manager
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_m2x_options_manager/it/

houzefa-abba committed:
[IMP] web_timeline: Safer demo view action override

Append our view_mode this way to avoid overwriting changes done by other
modules.

Chionne27 committed:
[FIX] web_m2x_options_manager: writing mode when create/edit wizard is false

OCA-git-bot committed:
[BOT] post-merge updates

ivantodorovich committed:
[ADD] web_widget_popover

MohamedOsman7 committed:
[IMP] web_chatter_position: improve form and chatter layout styles

Co-authored-by: Stefan Rijnhart <stefan@opener.amsterdam>

anusriNPS committed:
[16.0][ADD] web_sort_menu: Module web_sort_menu

anusriNPS committed:
[REF] web_company_color: Update fetching scss_template

   Using a function to fetch SCSS_TEMPLATE

jguenat committed:
[MIG] web_pwa_customize: Migration to 18.0

sergioM-S73 committed:
fix error: env.services.rpc is not a function

len-foss committed:
[MIG] web_dark_mode: Migration to 18.0

len-foss committed:
[IMP] web_dark_mode: pre-commit execution

DavidJForgeFlow committed:
[IMP] web_chatter_position: pre-commit fixes.

ljmnoonan committed:
[IMP] web_dark_mode Set dark theme to immitate VSCode Modern Dark
Also, change asset bundle loading from 'prepend' to 'before' This is in line with Odoo's own way of doing it.

ljmnoonan committed:
[IMP] web_dark_mode Change navbar from purple to dark
Also get rid of the bottom border.

ljmnoonan committed:
[IMP] web_dark_mode Fix tint and shade color functions

ljmnoonan committed:
[IMP] web_dark_mode Add lazy loading asset bundle
If you go to any activity view and hover over an empty cell you will notice that
without this fix the the on hover background of empty cells (.o_activity_empty_cell) set by
$component-active-bg which is itself set as mix($o-action, $o-gray-100, 20%) through $o-component-active-bg
completely ignores the overrides we set in primary_variables.dark.scss.
This is because the assets for that view are lazy loaded, so we need to include our override
bundles in the lazy bundle.

ljmnoonan committed:
[IMP] web_dark_mode Override BS variables
Just general customizations of BS variables to bring them in line with our style

ljmnoonan committed:
[IMP] web_dark_mode Improve notebook tab styling

AungKoKoLin1997 committed:
[IMP] web_form_banner: imp readme

OCA-git-bot committed:
Merge PR #3311 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update web_datetime_picker_default_time.pot

OCA-git-bot committed:
[BOT] post-merge updates

ljmnoonan committed:
[IMP] web_dark_mode Adjust tag and colorlist colors

ljmnoonan committed:
[IMP] web_dark_mode Adjust calendar coloring
Add secondary_variables override to get rid of colors that don't jive well with the dark background.
Unfortunately this does introduce an issue where if you have many calendar users, you can run out of colors
This issue exists in the enterprise dark mode as well, so hopefully it will be fixed.

ljmnoonan committed:
[IMP] web_dark_mode Adjust the styles of custom properties

DavidJForgeFlow committed:
[IMP] web_company_color: mobile systray button

When used on mobile/tablet view, the buttons and texts in the navbar do not match the company colors.

ljmnoonan committed:
[IMP] web_dark_mode Tweak the search bar
Set the background color to be darker and tone down the border around it.
The styling for the border is borrwed from the style of .o-mail-Composer-bg

ljmnoonan committed:
[IMP] web_dark_mode Set image background to white

ljmnoonan committed:
[IMP] web_dark_mode Tweak kanban controller
Set attachment image background to white and fix the column background when hovering with a draggable card

AungKoKoLin1997 committed:
[FIX] web_form_banner

- Fix not updating banner when pager is update or duplicate or create
- Fix presentation of help in web form banner rule

BhaveshHeliconia committed:
web_dialog_size: prevent export wizard from shrinking when dialog maximize is enabled

OCA-git-bot committed:
Merge PR #3302 into 18.0

Signed-off-by legalsylvain

Commit:
[UPD] Update web_editor_disable_chatgpt.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3319 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3317 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

carlos-lopez-tecnativa committed:
[IMP] web_responsive: adjust button styles and alignment in the chatter
The purpose is to reduce spacing so that all icons and buttons fit within the window without having to scroll.

ljmnoonan committed:
[IMP] web_theme_classic Refactor SCSS
Refactor all selectors to mirror the way odoo sets them. This results in a MUCH broader application of our styles that is not limited to form views.
Prefix all vars with "wtc-" in order to prevent them from overriding or being overridden by bootstrap vars.
Suffix all vars with !default to ensure that they can be overriden by dark mode.
Add $wtc-input-color-required to ensure text contrasts on $wtc-input-background-color-required. This is especially necessary in instances like the quantity and unit price columns of a sale order because by default these fields are set to a light blue.
Remove border on .note-editable, i.e. the HTML editor. This feature was broken and, as far as I am concerned, unnecessary.
Remove all button customization as it was all broken and is superceeded by OCA/web/web_save_discard_button anyways.

ljmnoonan committed:
[IMP] web_theme_classic Add dark mode assets
web_dark_mode is not merged yet.
https://github.com/OCA/web/pull/3324

mymage committed:
Added translation using Weblate (Italian)

ljmnoonan committed:
[IMP] web_dark_mode: Add myself to CONTRIBUTORS.md
I don't want to but my email on a public repo, so I just linked my github page instead.
I could not find any guideline against this, but please correct me if there is.

yankinmax committed:
[IMP] web_widget_popover: pre-commit auto fixes

yankinmax committed:
[18.0][MIG] web_widget_popover: Migration to 18.0

OCA-git-bot committed:
Merge PR #3248 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_pwa_customize.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3323 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

carlos-lopez-tecnativa committed:
[IMP] web_widget_product_label_section_and_note_name_visibility: Remove the incompatibility with the sale_product_configurator module in the ROADMAP.
This module was merged into sale in the following commit: https://github.com/odoo/odoo/commit/fdcd3d5804cb86f3bda472e093739b7393c63659

ljmnoonan committed:
[IMP] web_dark_mode Change thead active (column filter) bg for lists in forms
It was being set to $o-form-lightsecondary which I set lighter, so this tones it down a bit.

OCA-git-bot committed:
Merge PR #3342 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3308 into 18.0

Signed-off-by sebastienbeau

Commit:
[UPD] Update web_form_banner.pot

OCA-git-bot committed:
[BOT] post-merge updates

LauraCForgeFlow committed:
[ADD] web_widget_product_label_section_and_note_full_label

mymage committed:
Added translation using Weblate (Italian)

anusriNPS committed:
[MIG] web_sort_menu: Migration to 18.0

OCA-git-bot committed:
Merge PR #3337 into 18.0

Signed-off-by ivantodorovich

Commit:
[UPD] Update web_widget_popover.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (63 of 63 strings)

Translation: web-18.0/web-18.0-web_form_banner
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_form_banner/it/

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (31 of 31 strings)

Translation: web-18.0/web-18.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_responsive/it/

SilvioC2C committed:
[IMP] web_m2x_options_manager: pre-commit auto fixes

SilvioC2C committed:
[MIG] web_m2x_options_manager: Migration to 17.0

Field ``option_create_edit_wizard`` has been dropped in the migration: the pop-up wizard does not exist anymore, so the field has no use

OCA-git-bot committed:
Merge PR #3272 into 18.0

Signed-off-by hbrunn

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: web-18.0/web-18.0-web_widget_popover
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_popover/it/

Ryoko04 committed:
Added translation using Weblate (Japanese)

Ryoko04 committed:
Translated using Weblate (Japanese)

Currently translated at 100.0% (2 of 2 strings)

Translation: web-18.0/web-18.0-web_m2x_options
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_m2x_options/ja/

OCA-git-bot committed:
Merge PR #3264 into 18.0

Signed-off-by hbrunn

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3290 into 18.0

Signed-off-by etobella

Commit:
[UPD] Update web_chatter_position.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/

Xino61122 committed:
[IMP] web_touchscreen: pre-commit auto fixes

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/it/

OCA-git-bot committed:
Merge PR #3344 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_widget_product_label_section_and_note_full_label.pot

OCA-git-bot committed:
[BOT] post-merge updates

Xino61122 committed:
[MIG][18.0] web_touchscreen: Migration to 18

MT-11823

Commit:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
Merge PR #3322 into 18.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

SilvioC2C committed:
[IMP] web_m2x_options_manager: pre-commit auto fixes

SilvioC2C committed:
[MIG] web_m2x_options_manager: Migration to 18.0

OCA-git-bot committed:
Merge PR #3361 into 18.0

Signed-off-by tarteo

Commit:
[UPD] Update web_m2x_options_manager.pot

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

OCA-git-bot committed:
Merge PR #3365 into 18.0

Signed-off-by yajo

Commit:
[UPD] Update web_touchscreen.pot

OCA-git-bot committed:
[BOT] post-merge updates

Saran440 committed:
[FIX] web_save_discard_button: readme

OCA-git-bot committed:
Merge PR #3390 into 18.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[IMP] web_form_banner: adj to resolve eslint warnings

OCA-git-bot committed:
Merge PR #3345 into 18.0

Signed-off-by ivs-cetmix

Commit:
[UPD] Update web_sort_menu.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3324 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_dark_mode.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

RicardCForgeFlow committed:
[18.0][FIX] web_widget_x2many_2d_matrix: fix sticky header/column overlap

OCA-git-bot committed:
Merge PR #3404 into 18.0

Signed-off-by legalsylvain

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (21 of 21 strings)

Translation: web-18.0/web-18.0-web_pivot_computed_measure
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_pivot_computed_measure/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (8 of 8 strings)

Translation: web-18.0/web-18.0-web_widget_x2many_2d_matrix
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_x2many_2d_matrix/it/

LauraCForgeFlow committed:
[IMP] web_widget_one2many_tree_line_duplicate: make duplicate icon hideable

LauraCForgeFlow committed:
[ADD] web_tree_column_keyboard_resize

RicardCForgeFlow committed:
[18.0][FIX] web_widget_x2many_2d_matrix: fix total updates

Totals were not consistently refreshing immediately on cell edits.

- Replace 'onWillUpdateProps' for 'onWillRender'.
  This ensures totals update immediately when a cell value changes.
- Expand the JS test with a case verifying totals after
  multiple consecutive edits.
- Refine JS test filtering in Python HttpCase to target only this module.

OCA-git-bot committed:
Merge PR #3412 into 18.0

Signed-off-by legalsylvain

Commit:
[UPD] Update web_tree_column_keyboard_resize.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3409 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_widget_one2many_tree_line_duplicate.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_widget_one2many_tree_line_duplicate
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_one2many_tree_line_duplicate/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: web-18.0/web-18.0-web_widget_one2many_tree_line_duplicate
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_one2many_tree_line_duplicate/it/

OCA-git-bot committed:
Merge PR #3391 into 18.0

Signed-off-by ivs-cetmix

OCA-git-bot committed:
[BOT] post-merge updates

hbrunn committed:
[IMP] update dotfiles

pedrobaeza committed:
Merge pull request #3416 from hbrunn/18.0-copier-update-2026-01-23

[18.0][IMP] update dotfiles

OCA-git-bot committed:
Merge PR #3413 into 18.0

Signed-off-by hbrunn

OCA-git-bot committed:
[BOT] post-merge updates

carlos-lopez-tecnativa committed:
[FIX] web_timeline: Render the SVG correctly to draw dependencies properly

This issue was introduced during the migration to v18 when converting old jQuery code to native JavaScript in this commit: https://github.com/OCA/web/commit/258a5a92dc85fa5cc6dd9181ca444e3b4b290d0f

Fixes https://github.com/OCA/web/issues/3420

OCA-git-bot committed:
Merge PR #3424 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

SirPyTech committed:
[IMP] web_company_color: Compile SCSS

Otherwise function calls are not evaluated in the created attachment

OCA-git-bot committed:
Merge PR #3426 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #3427 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

remi-filament committed:
[18.0][FIX] web_widget_x2many_2d_matrix: monetary field display

remi-filament committed:
[IMP] hide input border on monetary fields

ZubikIT committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (31 of 31 strings)

Translation: web-18.0/web-18.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_responsive/ru/

OCA-git-bot committed:
Merge PR #3432 into 18.0

Signed-off-by hbrunn

OCA-git-bot committed:
[BOT] post-merge updates

SirPyTech committed:
[IMP] web_timeline: Allow to select in action's view_ids

OCA-git-bot committed:
Merge PR #3055 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_time_range_menu_custom.pot

OCA-git-bot committed:
[BOT] post-merge updates

arch-fan committed:
[ADD] web_toggle_chatter

OCA-git-bot committed:
Merge PR #3444 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_toggle_chatter.pot

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

mymage committed:
Added translation using Weblate (Italian)

SirPyTech committed:
[REM] web_timeline: Maintainer

OCA-git-bot committed:
Merge PR #3460 into 18.0

Signed-off-by tarteo

OCA-git-bot committed:
[BOT] post-merge updates

dmoranp committed:
[FIX] web_responsive: remove invalid hotkey prop from DropdownItem

In Odoo 18.0, the DropdownItem component no longer accepts a hotkey
prop. Passing it causes an OWL crash:

  Invalid props for component 'DropdownItem': unknown key 'hotkey'

This error appears in mobile mode when accessing form views that have
status bar action buttons (e.g. scheduled actions).

The template extension only added the invalid hotkey attribute and had
no other content, so the file is removed entirely.

Fixes https://github.com/OCA/web/issues/3467

OCA-git-bot committed:
Merge PR #3473 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

bo-eska committed:
Added translation using Weblate (Turkish)

OCA-git-bot committed:
Merge PR #3483 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

hbrunn committed:
[REF] web_company_color: simplify write code

hbrunn committed:
[IMP] web_company_color: allow to set bottom border color of navbar

hbrunn committed:
[IMP] web_company_color: test ignore_company_color context key

bo-eska committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (19 of 19 strings)

Translation: web-18.0/web-18.0-web_notify
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_notify/tr/

OCA-git-bot committed:
Merge PR #3484 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update web_company_color.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-18.0/web-18.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_company_color/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-18.0/web-18.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_company_color/it/

StefanRijnhart committed:
[FIX] checklog-odoo.cfg: ignore Killing chrome descendants-or-self warning

Warning introduced in https://github.com/odoo/odoo/commit/d22e0b4e181de0940810b0f0a760bcfe288042a7. The previous warning apparently did not occur, as it was not featured in checklog-odoo.cfg.

```
2026-04-13 03:20:41,702 367 WARNING odoo odoo.addons.web_disable_export_group.tests.test_tour.TestTour.test_admin: Killing chrome descendants-or-self of 382: 6 remaining
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
- chrome (zombie)
```

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (5 of 5 strings)

Translation: web-18.0/web-18.0-web_chatter_position
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_chatter_position/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (3 of 3 strings)

Translation: web-18.0/web-18.0-web_widget_dropdown_dynamic
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_dropdown_dynamic/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (8 of 8 strings)

Translation: web-18.0/web-18.0-web_widget_x2many_2d_matrix
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_widget_x2many_2d_matrix/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (19 of 19 strings)

Translation: web-18.0/web-18.0-web_notify
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_notify/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (16 of 16 strings)

Translation: web-18.0/web-18.0-web_timeline
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_timeline/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: web-18.0/web-18.0-web_time_range_menu_custom
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_time_range_menu_custom/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (18 of 18 strings)

Translation: web-18.0/web-18.0-web_company_color
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_company_color/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (21 of 21 strings)

Translation: web-18.0/web-18.0-web_pivot_computed_measure
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_pivot_computed_measure/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (20 of 20 strings)

Translation: web-18.0/web-18.0-web_ir_actions_act_multi
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_ir_actions_act_multi/es/

Ed-Spain committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (42 of 42 strings)

Translation: web-18.0/web-18.0-web_quick_start_screen
Translate-URL: https://translation.odoo-community.org/projects/web-18-0/web-18-0-web_quick_start_screen/es/

pedrobaeza committed:
Merge pull request #3507 from StefanRijnhart/18.0-checklog-odoo.cfg-ignore_warning_killing_chrome

[18.0][FIX] checklog-odoo.cfg: ignore Killing chrome descendants-or-self warning
